config-common/src/super_config.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+// TODO: Move this to aziotctl-common
+
+/// This is the config stored in `/etc/aziot/config.toml`
+///
+/// It is an amalgam of the individual services' configs, with some tweaks:
+///
+/// - Wherever key IDs were used to link keys between services, the key is referenced directly.
+///   For example, instead of IS provisioning referencing a symmetric key by its key ID and KS optionally associating a key ID with a key URI,
+///   the IS provisioning optionally references the key URI directly.
+///
+/// - Wherever cert IDs were used to link certs between services, the key is referenced directly.
+///   For example, instead of IS provisioning referencing a cert by its cert ID and CS associating a cert ID with a cert URI or an issuance method,
+///   the IS provisioning references the cert URI or issuance method directly.
+///
+/// - A consequence of the above is that the EST endpoint spec only allows one of identity or bootstrap identity X.509 auth to be specified, not both.
+///   (If bootstrap identity was specified, the regular identity would be IDs.)
+///
+/// Unfortunately it's not easy to do this without duplicating all the individual services' config types.
+/// Inner types are reused when possible, ie when they don't need modification to work with the super-config.
```

Comment:
What if we use g e n e r i c s to to avoid some of the more blatant duplication.

On one hand, it's a bit overkill, but on the other hand, it will guarantee that things are kept in-sync if/when configuration types start getting changed.

Reply:
The guarantee already exists, because the translation code in `apply` will fail if either the super-config or the individual service config is changed. (That's why I ensure all the code uses destructuring instead of field access, doesn't use `..` in pats, etc.)

I did think of generics, but some of the types like the `Est` tree are sufficiently different that it may not work. I'll check again.